### PR TITLE
Upgrade Ember and Ember CLI

### PR DESCRIPTION
**Pull request by blimmer**

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -30,5 +30,5 @@
   "white": false,
   "eqnull": true,
   "esnext": true,
-  "unused": "vars"
+  "unused": true
```
Comment (samselikoff) [inline .jshintrc]: can we keep this `vars`?

File: .npmignore [hunk 2]
```diff
@@ -1,10 +1,10 @@
 .sass-cache/
 _site/
 bower_components/
-jekyll-tmp/
 tests/
 tmp/
 dist/
+jekyll-tmp/
```
Comment (samselikoff) [inline .npmignore]: this is for the gh-pages site, ok to keep?

Comment (blimmer) [inline .npmignore]: Yep, it's still there. My editor must have moved it down.

File: .jshintrc (hunk 1 above)
Comment (blimmer) [inline .jshintrc]: Yep, was just matching ember-cli standards, but I think it's totally acceptable to do this.

**Comment by blimmer** [comment]
We probably shouldn't merge this until we figure out what changed in Ember-CLI that [caused the node tests to break](https://travis-ci.org/samselikoff/ember-cli-mirage/jobs/75733531#L1026).

It looks like calling `EmberAddon()` without any options now throws an exception:

```
TypeError: Cannot read property 'project' of undefined
      at getMirageAddon (tests/unit/addon-tree-test-node.js:7:18)
      at Context.<anonymous> (tests/unit/addon-tree-test-node.js:30:23)
```

**Comment by samselikoff** [comment]
started to debug this but, the first Addon#treeFor test isn't even passing for me on master. Do all tests on master pass for you?

**Comment by blimmer** [comment]
all except the first node test work on master for me:
```
  Addon
    #treeFor
      1) returns an empty tree in production environment by default //fails
```

**Comment by samselikoff** [comment]
ditto. it's passing on travis tho, strange

**Comment by blimmer** [comment]
Still looking into this. It looks like with the new `ember-cli-build.js` file, it gets passed a set of defaults (which were previously "just there"). I am trying to track down how those get passed in because that should fix the problem that exists here.

File: tests/unit/addon-tree-test-node.js [hunk 3]
```diff
@@ -29,7 +35,11 @@ describe('Addon', function() {
       process.env.EMBER_ENV = 'production';
       var addonTree = getMirageAddon().treeFor('addon');
 
-      expect(addonTree.inputTrees.length).to.be.equal(0);
+      if (addonTree._inputNodes) {
+        expect(addonTree._inputNodes.length).to.equal(0);
+      } else {
+        expect(addonTree._inputNodes).to.be.undefined;
+      }
```
Comment (blimmer) [inline tests/unit/addon-tree-test-node.js]: This is very strange, but goes along with the behavior you and I were seeing early on master (where `length` was undefined on Mac, but not on Travis' Ubuntu version).

**Comment by blimmer** [comment]
@samselikoff this is passing now.

**Comment by samselikoff** [comment]
awesome, looks great!

**Comment by samselikoff** [comment]
could you squash? makes it easier when updating the changelog if I can look back through history quickly. (I'm fine with multiple commits on more complex PRs)

**Comment by blimmer** [comment]
Makes sense

**Comment by samselikoff** [comment]
perfect, ty!

**Pull request merged**